### PR TITLE
order checkout flow and views

### PR DESCRIPTION
**Pull request by Mikearaya**
on this draft pull request, I'm adding the order list page based on the current users along with the details page associated with each order. 

I'm following the existing folder and page structure to make a uniform and understandable codebase :smile:  

### please do let me know if you have any suggestion or feedback regarding the changes I'm making 

**Thank you** :bowing_man:

### CODE REVIEW

**Comment by schmidsi** [comment]
Makes sense with the reorganisation of the code!

File: storefront/modules/orders/hooks/useOrderDetailQuery.jsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { useQuery } from '@apollo/react-hooks';
+
+import OrderFragment from '../fragments/OrderFragment';
+import OrderItemFragment from '../fragments/OrderItemFragment';
+
+const OrderDetailQuery = gql`
+  query OrderDetailQuery($orderNumber: ID!) {
+    order(orderId: $orderNumber) {
+      ...OrderFragment
+      ...OrderItemFragment
```
Comment (schmidsi) [inline storefront/modules/orders/hooks/useOrderDetailQuery.jsx:11]:
```gql
items {
   ...OrderItemFragment
}
```

File: storefront/modules/orders/fragments/OrderItemFragment.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import gql from 'graphql-tag';
+
+const OrderItemFragment = gql`
+  fragment OrderItemFragment on Order {
```
Comment (schmidsi) [inline storefront/modules/orders/fragments/OrderItemFragment.tsx:4]:
```gql
fragment OrderItemFragment on OrderItem
```

File: storefront/pages/bezahlen.tsx [hunk 3]
```diff
@@ -226,6 +229,35 @@ const BillingSection = () => {
...
+      orderId: user.cart._id,
+      deliveryProviderId: deliveryProvider._id,
+    });
+
+    checkOutCart({
+      orderId: user.cart._id,
+      orderContext: { status: 'CONFIRMED' },
+      paymentContext: { status: 'PAID' },
+      deliveryContext: { status: 'OPEN' },
+    });
```
Comment (schmidsi) [inline storefront/pages/bezahlen.tsx:259]: You need to redirect after the checkout

File: storefront/pages/bezahlen.tsx [hunk 4]
```diff
@@ -226,6 +229,35 @@ const BillingSection = () => {
 };
 
 const Payment = () => {
+  const { user } = useUserQuery();
+  const { setOrderPaymentProvider } = useSetOrderPaymentProviderMutation();
+  const { setOrderDeliveryProvider } = useSetOrderDeliveryProviderMutation();
+  const { checkOutCart } = useCheckOutCartMutation();
+
+  const handleCheckout = () => {
```
Comment (schmidsi) [inline storefront/pages/bezahlen.tsx:237]: This needs to be an async function: 
```
const handleCheckout = async () => {
```

File: storefront/pages/bezahlen.tsx [hunk 5]
```diff
@@ -226,6 +229,35 @@ const BillingSection = () => {
...
+    const paymentProvider = user.cart.supportedPaymentProviders.find(
+      ({ type }) => type === 'INVOICE',
+    );
+
+    const deliveryProvider = user.cart.supportedDeliveryProviders[0];
+
+    setOrderPaymentProvider({
```
Comment (schmidsi) [inline storefront/pages/bezahlen.tsx:244]: So you can await these:

```ts
await setOrderPaymentProvider({
```

File: storefront/pages/bezahlen.tsx [hunk 6]
```diff
@@ -226,6 +229,35 @@ const BillingSection = () => {
...
+      orderId: user.cart._id,
+      paymentProviderId: paymentProvider._id,
+    });
+
+    setOrderDeliveryProvider({
```
Comment (schmidsi) [inline storefront/pages/bezahlen.tsx:249]: Same here: `await`

File: storefront/pages/bezahlen.tsx [hunk 7]
```diff
@@ -226,6 +229,35 @@ const BillingSection = () => {
...
+      orderId: user.cart._id,
+      deliveryProviderId: deliveryProvider._id,
+    });
+
+    checkOutCart({
```
Comment (schmidsi) [inline storefront/pages/bezahlen.tsx:254]: Same here: `await`

File: storefront/modules/orders/fragments/OrderFragment.tsx [hunk 8]
```diff
@@ -5,11 +5,43 @@ const OrderFragment = gql`
     _id
     status
     created
+    updated
+    ordered
+    country {
+      flagEmoji
```
Comment (schmidsi) [inline storefront/modules/orders/fragments/OrderFragment.tsx:11]: This is cool 😎

File: storefront/modules/orders/hooks/useUserOrderListQuery.tsx [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import gql from 'graphql-tag';
+import { useQuery } from '@apollo/react-hooks';
+
+import OrderFragment from '../fragments/OrderFragment';
+
+const UserOrderListQuery = gql`
+  {
+    me {
```
Comment (schmidsi) [inline storefront/modules/orders/hooks/useUserOrderListQuery.tsx:8]: add
```gql
_id
```

File: storefront/pages/order/[id].tsx [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+          <div className="col-sm-6">
+            <div className="row">
+              {order.items.map((item, i) => (
+                <div className="col-sm-6" key={i}>
+                  <h3 className="px-2 my-3">{item?.product?.texts?.title}</h3>
+                  <img src={getProductMediaUrl(item.product)} />
+                  <div className="p-2">
+                    <h4 className="my-0">
+                      {item?.total.currency}{' '}
+                      {item?.total.amount * item?.quantity}
```
Comment (schmidsi) [inline storefront/pages/order/[id].tsx:57]: Unnecessary duplication of the multiplication: The total already multiplies quantity with unitPrice.

File: storefront/pages/order/[id].tsx [hunk 11]
```diff
@@ -0,0 +1,74 @@
...
+import useOrderDetailQuery from '../../modules/orders/hooks/useOrderDetailQuery';
+
+const OrderDetail = () => {
+  const router = useRouter();
+
+  const { order, loading } = useOrderDetailQuery({ orderId: router.query.id });
+  if (loading) return <p>loading .... </p>;
+  console.log(order.items);
+
+  return (
```
Comment (schmidsi) [inline storefront/pages/order/[id].tsx:15]: Reuse as much as possible from here: 

https://github.com/unchainedshop/currybag-website/blob/master/storefront/pages/warenkorb.tsx

Comment (schmidsi) [reply storefront/pages/order/[id].tsx:15]: And here: https://github.com/unchainedshop/currybag-website/blob/master/storefront/modules/cart/components/ManageCart.tsx

File: storefront/modules/orders/hooks/useUserOrderListQuery.tsx (hunk 9 above)
Comment (Mikearaya) [reply storefront/modules/orders/hooks/useUserOrderListQuery.tsx:8]: **Fixed**

File: storefront/pages/order/[id].tsx (hunk 10 above)
Comment (Mikearaya) [reply storefront/pages/order/[id].tsx:57]: **Fixed** I had removed it before didn't commit

File: storefront/modules/orders/components/order.module.css [hunk 12]
```diff
@@ -0,0 +1,133 @@
+table#order {
```
Comment (schmidsi) [inline storefront/modules/orders/components/order.module.css:1]: @fliptation Maybe you have some feedback about CSS authoring?

Comment (Mikearaya) [reply storefront/modules/orders/components/order.module.css]: @fliptation CSS updated and also the order button only visible for users with orders, in addition, to be logged in.

**Pull request merged**